### PR TITLE
Compose message feature

### PR DESCRIPTION
**Pull request by johannaanden**
## Pivotal Tracker link
[Link to PT](https://www.pivotaltracker.com/story/show/162084273)

## Actions taken
Created a feature file -> compose_message.feature
Created a step definition file -> compose_message_steps.rb
Added feature, user story and scenarios
Refactor the code

## What we have learned
How to set up feature and run tests
Refactor the code

### CODE REVIEW

File: features/compose_message.feature [hunk 1]
```diff
@@ -0,0 +1,42 @@
+Feature: Compose message
+
+    As a User
+    In order to send a message to someone
+    I want to be able to compose a message
+
+Background:
+    Given following users exists
+        |name   |email              |password   |password_confirmation |
+        |Jenny  |jenny@random.com|password   |password               |
```
Comment (oliverochman) [inline features/compose_message.feature]: Please format the table

File: features/compose_message.feature [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        |Daniel|daniel@random.com|password|password|
+
+Scenario: Composing a message (Happy Path)
+    Given I am logged in as "Jenny"
+    And I am on the "inbox" page
+    Then I should see "Compose"
```
Comment (oliverochman) [inline features/compose_message.feature]: If you are able to click the button, it is there. No need for this test

File: features/compose_message.feature [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    When I click "Compose"
+    And I select "Daniel" from the "Recipients"
+    And I fill in "Subject" with "Hello"
+    And I fill in "conversation_body" with "World"
+    When I click "Message"
+    Then I should see "Your message was successfully sent!"
+
+Scenario: Composing a message (Sad Path - No recipients)
+    Given I am logged in as "Jenny"
```
Comment (oliverochman) [inline features/compose_message.feature]: The steps on line 25 and 26 are used in every scenario, move it to the background

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: features/step_definitions/compose_message_steps.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        FactoryBot.create(:user, user)
+    end
+end
+
+Given('I am logged in as {string}') do |name|
+  user = User.find_by(name: name)
+  login_as(user, scope: :user)
+end
+
+Given("I am on the {string} page") do |value|
```
Comment (oliverochman) [inline features/step_definitions/compose_message_steps.rb]: You are not using the value here

Comment (oliverochman) [reply features/step_definitions/compose_message_steps.rb]: its hardcoded

File: features/compose_message.feature (hunk 1 above)
Comment (johannaanden) [reply features/compose_message.feature]: Done

File: features/compose_message.feature (hunk 2 above)
Comment (johannaanden) [reply features/compose_message.feature]: Done

File: features/compose_message.feature (hunk 3 above)
Comment (johannaanden) [reply features/compose_message.feature]: Done

File: features/step_definitions/compose_message_steps.rb (hunk 4 above)
Comment (johannaanden) [reply features/step_definitions/compose_message_steps.rb]: Done

**Review by oliverochman — APPROVED** [review]

**Pull request merged**